rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/util/DefaultImageNameFactory.groovy

```diff
@@ -39,35 +39,43 @@ public class DefaultImageNameFactory implements ImageNameFactory {
     def timestamp = clock.millis()
 
     List<String> packageNameList = bakeRequest?.package_name?.tokenize(" ")
-    def firstPackageName
     PackageNameConverter.OsPackageName osPackageName
     String appVersionStr
     AppVersion appVersion
+    def baseImagePackageName
 
     if (packageNameList) {
-      // For now, we only take into account the first package name when generating the appversion tag.
-      firstPackageName = packageNameList[0]
+      packageNameList.eachWithIndex { packageName, index ->
 
-      // Passing in firstPackageName here to avoid tokenizing the package name list twice.
-      osPackageName = PackageNameConverter.buildOsPackageName(selectedOptions.baseImage.packageType, firstPackageName)
-      appVersionStr = PackageNameConverter.buildAppVersionStr(bakeRequest, osPackageName)
-      appVersion = AppVersion.parseName(appVersionStr)
+        osPackageName = PackageNameConverter.buildOsPackageName(selectedOptions.baseImage.packageType, packageName)
```

Comment:
Actually, maybe we shouldn't even capture the response from `AppVersion.parseName(appVersionStr)` in a variable at all?